### PR TITLE
feature: add schema to code

### PR DESCRIPTION
**Pull request by TomBurdge**
Schema to code add.

Adds an optional `prepend_pl` flag, as it's common to `import polars as pl`, and then you don't have to worry about importing the right types.

### CODE REVIEW

File: tests/test_schema_helpers.py [hunk 1]
```diff
@@ -0,0 +1,16 @@
+from harley.schema_helpers import print_schema_as_code
+from typing import OrderedDict
+import polars.datatypes as T
+import pytest
+from io import StringIO
+import sys
+
+@pytest.mark.parametrize("prepend_pl, exp", [(True, 'OrderedDict([("col_1", pl.String),("col_2", pl.Array),])'),(False,'OrderedDict([("col_1", String),("col_2", Array),])')])
+def test_print_schema_as_code(prepend_pl:bool, exp:str):
```
Comment (LyndonFan) [inline tests/test_schema_helpers.py]: Looks good, can we check it works with different number of columns? Eg 1, 3, or even 0 (not sure if it's possible with polars 🤷)

File: harley/schema_helpers.py [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+    Optional `prepend_pl` flag for prepending 'pl.' to the column types.
+    """
+    if prepend_pl:
+        schema = OrderedDict(
+        [
+            (field, "pl."+ str(d_type))
+            for field, d_type in schema.items()
+        ]
+    )
+    out = "OrderedDict(["
```
Comment (LyndonFan) [inline harley/schema_helpers.py:16]: Could just use `str(schema)`? Or is this to ensure we have `"..."` for strings?

File: harley/schema_helpers.py [hunk 3]
```diff
@@ -0,0 +1,20 @@
+from typing import OrderedDict
+from polars import DataType
+
+def print_schema_as_code(schema: OrderedDict[str, DataType], prepend_pl: bool = False)-> str:
```
Comment (LyndonFan) [inline harley/schema_helpers.py]: Might want to be `-> None` as it's printing the result

**Review by LyndonFan — CHANGES_REQUESTED** [review]

File: harley/schema_helpers.py (hunk 2 above)
Comment (TomBurdge) [reply harley/schema_helpers.py:16]: Yeah a tiny thing, and probably not worth it but `ruff`/`black` will format strings to double quotes so may as well do it the "recommended" way.

File: harley/schema_helpers.py (hunk 3 above)
Comment (TomBurdge) [reply harley/schema_helpers.py]: Good shout, implemented in latest commits.

File: tests/test_schema_helpers.py (hunk 1 above)
Comment (TomBurdge) [reply tests/test_schema_helpers.py]: Add tests for 1 and 3 columns.

File: harley/schema_helpers.py (hunk 2 above)
Comment (LyndonFan) [reply harley/schema_helpers.py:16]: Possibly. But what about nested schemas? (e.g. with list / struct)
```python
data = [{"name":{"first_name":"Charles","last_name":"Spring"},"age":18,"nicknames":["Charlie","Char"]},{"name":{"first_name":"Nicholas","last_name":"Nelson"},"age":19,"nicknames":["Nick"]}]
df = pl.DataFrame(data)
print(df.schema)
# Schema([('name', Struct({'first_name': String, 'last_name': String})), ('age', Int64), ('nicknames', List(String))]
harley.print_schem_as_code(df.schema, prepend_pl=True)
# OrderedDict([("name", pl.Struct({'first_name': String, 'last_name': String})),("age", pl.Int64),("nicknames", pl.List(String)),])
```

**Review by LyndonFan — CHANGES_REQUESTED** [review]
Possibly missing changes in `__init__.py`?

**Comment by TomBurdge** [comment]
> Possibly missing changes in `__init__.py`?

Added in latest commit

**Review by LyndonFan — APPROVED** [review]

**Pull request merged**